File: runtime/ruby/units/wasi/args_sizes_get.rb
# requires: memory/i32_store
def wasi_args_sizes_get(argc_ptr, buf_size_ptr)
  @memory.i32_store(argc_ptr, @args.size)
  @memory.i32_store(buf_size_ptr, @args.sum { |a| a.bytesize + 1 })
  ERRNO_SUCCESS
end

File: runtime/python/units/wasi/args_sizes_get.py
# requires: memory/iws
def wasi_args_sizes_get(self, argc_ptr, buf_size_ptr):
    self.memory.iws(argc_ptr, len(self.args))
    self.memory.iws(buf_size_ptr, sum(len(a) + 1 for a in self.args))
    return self.ERRNO_SUCCESS
